bytes/bytes/raw/file_raw_repository.py

```diff
@@ -62,3 +72,68 @@ def _raw_file_path(self, raw_id: UUID, boefje_meta: BoefjeMeta) -> Path:
 
     def _index(self, raw_id: UUID) -> str:
         return str(raw_id)[: self.UUID_INDEX]
+
+
+class S3RawRepository(RawRepository):
+    def __init__(
+        self,
+        file_middleware: FileMiddleware,
+        bucket_per_org: bool,
+        s3_bucket_prefix: str | None = "OpenKAT-",
+        s3_bucket_name: str | None = "OpenKAT",
+    ) -> None:
+        self.file_middleware = file_middleware
+        self.bucket_per_org = bucket_per_org
+        self.s3_bucket_prefix = s3_bucket_prefix
+        self.s3_bucket_name = s3_bucket_name
+
+        self._session: Session = None
+        self._s3resource = None
+
+    @property
+    def s3resource(self):
+        if self._s3resource:
+            return self._s3resource
+        self._session = BotoSession()
+        self._s3resource = self._session.resource("s3")
+        return self._s3resource
+
+    def get_or_create_bucket(self, organization: str):
+        # Create a bucket, and if it exists already return that instead
+        bucket_name = self.s3_bucket_name
+        if self.bucket_per_org:
+            bucket_name = f"{self.s3_bucket_prefix}{organization}"
+            try:
+                bucket = self.s3resource.create_bucket(Bucket=bucket_name)
```

Comment:
I think we should keep using the method in case `s3resource` does not exist yet

Reply:
While I understand why this is implemented as such, we should avoid using properties for instantiating such resources. Also, using this method (`get_or_create_bucket`) without a resource is a misuse of the method and should naturally yield in an error.

If you need this resource to be available when using the method, consider instantiating it in the `S3RawRepository` instantiation

Reply:
You are totally correct. There is no need to have a method cache this property when it can just be a property that is assigned with the `S3RawRepository` instantiation

Reply:
Agreed. I believe instantiating the S3 connection shouldn't be expensive since it's just an HTTP API client. It wil usually only establish a connection when an actual request is made, so there shouldn't be much overhead in instantiation/ resources. Now that I think of this, it's probably even safe to instantiate it without needing to cache it